dls-controls/malcolmjs
# Feature/prevent link overwrites #210

## Description

Prevented links being overwritten if the input port is already connected.
Updated the ports so they have a colour based on their type (using the top-level theme object).

## Testing instructions

Add a set up instructions describing how the reviewer should test the code
- Review code
- Check Travis build
- Review changes to test coverage
- Run the site, go to the layout view, look at the port colours, try to make a link to an existing input and see that it doesn't happen.

## Agile board tracking

connect to #210 
connect to #273 
closes #210 
closes #273
